fs/driver/fs_registerdriver.c

```diff
@@ -67,33 +125,69 @@ int register_driver(FAR const char *path,
                     mode_t mode, FAR void *priv)
 {
   FAR struct inode *node;
-  int ret;
+  int               ret;
 
-  sched_note_mark(NOTE_TAG_DRIVERS, path);
+  ret = lock_populate_inode(path, fops, mode, priv, &node);
 
-  /* Insert a dummy node -- we need to hold the inode semaphore because we
-   * will have a momentarily bad structure.
-   */
+  if (ret == OK)
+    {
+      inode_unlock();
+#ifdef CONFIG_FS_NOTIFY
+      notify_create(path);
+#endif
+    }
 
-  inode_lock();
-  ret = inode_reserve(path, mode, &node);
-  if (ret >= 0)
+  return ret;
+}
+
+#ifdef CONFIG_EEPROM
+
+/****************************************************************************
+ * Name: register_eepromdriver
+ *
+ * Description:
+ *   Register an EEPROM character driver inode into the pseudo file system.
+ *
+ * Input Parameters:
+ *   path - The path to the inode to create
+ *   fops - The file operations structure
+ *   mode - inmode privileges
+ *   priv - Private, user data that will be associated with the inode.
+ *   size - EEPROM size in bytes
+ *
+ * Returned Value:
+ *   Zero on success (with the inode point in 'inode'); A negated errno
+ *   value is returned on a failure (all error values returned by
+ *   inode_reserve):
+ *
+ *   EINVAL - 'path' is invalid for this operation
+ *   EEXIST - An inode already exists at 'path'
+ *   ENOMEM - Failed to allocate in-memory resources for the operation
+ *
+ ****************************************************************************/
+
+int register_eepromdriver(FAR const char *path,
```

Comment:
why not forward `register_driver(path, fops, mode, priv)` to `register_eepromdriver(path, fops, mode, priv, 0)` and remove lock_populate_inode?

Reply:
`register_eepromdriver` was hidden inside `#ifdef CONFIG_EEPROM`. I've updated now that the i_size field is always declared.